frontend/model/contracts/mailbox.js

```diff
@@ -98,63 +127,108 @@ sbp('chelonia/defineContract', {
         username: string,
         contractID: optional(string)
       }),
-      process ({ meta, data }, { state }) {
-        if (state.attributes.creator === data.username) {
-          if (state.users[meta.username]) {
-            throw new TypeError(L('Already existing direct message channel.'))
-          }
-        } else if (state.attributes.creator === meta.username) {
-          if (!state.users[data.username] || state.users[data.username].joinedDate) {
+      process ({ meta, data }, { state, getters }) {
+        const me = state.attributes.creator
+        if (me !== meta.username && getters.ourOneToOneDirectMessages[data.username]) {
+          throw new TypeError(L('Already existing direct message channel.'))
+        } else if (me === meta.username) {
+          if (!getters.ourOneToOneDirectMessages[data.username] || getters.ourOneToOneDirectMessages[data.username].joinedDate) {
             throw new TypeError(L('Never created or already joined direct message channel.'))
           }
-        } else {
-          throw new TypeError(L('Incorrect mailbox creator to join direct message channel.'))
         }
-        const joinedDate = state.attributes.autoJoinAllowance ? meta.createdDate : null
-        if (state.attributes.creator === data.username) {
-          Vue.set(state.users, meta.username, {
-            contractID: data.contractID,
-            creator: meta.username,
+
+        const joinedDate = !state.attributes.autoJoinAllowance && me !== meta.username ? null : meta.createdDate
+        if (me !== meta.username) {
+          Vue.set(state.chatRooms, data.contractID, {
+            partner: data.username,
             hidden: false,
             joinedDate
           })
         } else {
-          Vue.set(state.users[data.username], 'joinedDate', joinedDate)
+          const contractID = getters.ourOneToOneDirectMessages[data.username].contractID
+          Vue.set(state.chatRooms[contractID], 'joinedDate', joinedDate)
```

Comment:
This isn't making much sense to me... I thought we agreed that DMs and GMs would be merged into the same thing?

Why then do we have a special getter for `ourOneToOneDirectMessages` and a separate action `'gi.actions/mailbox/createGroupMessage'`?

Notice in Slack 1-on-1 DMs can be turned into multi-user GMs:

![Screen Shot 2023-02-14 at 5 11 02 PM](https://user-images.githubusercontent.com/138706/218899458-7990aa38-ba88-4494-a2dd-aa37e22ccd56.jpg)

So we can simplify our code and treat all DMs as GMs, right?

Reply:
@taoeffect, as you described in Slack, we could add a list of users in the dms in mailbox contract state.
```
{ dms: { [chatroomID1]: { hidden: false, users: [ user1contractID, user2contractID, user3contractID ] } } }
```
While I am trying to do this, I found an issue. Let's imagine the following situation.
If someone create a direct message and add 4 people (so 5 people totally), 5 people would sync the chatroom contract together and 5 mailbox contracts should be up-to-date to the chatroom contract states (at least users inside).
At this moment, if anyone is added to this direct message (so totally 6 people), a new people would sync the chatroom contract and create a new `dms` field in his mailbox contract and the previous 5 people should update their mailbox contracts because there is a `users` field inside. I think this is a not good in terms of efficiency.

I wanted to design the mailbox contract that wouldn't be reactive to any of the chatroom contract state. So the problem is that I want to remove `users` field in the `dms` of mailbox contract. I will try to find a solution to merge `createDirectMessage` and `createGroupMessage`.